web3_proxy/src/frontend/authorization.rs

```diff
@@ -1084,6 +1098,39 @@ impl Web3ProxyApp {
         }
     }
 
+    /// Get the balance for the user.
+    ///
+    /// If a subuser calls this function, the subuser needs to have first attained the user_id that the rpc key belongs to.
+    /// This function should be called anywhere where balance is required (i.e. only rpc calls, I believe ... non-rpc calls don't really require balance)
+    pub(crate) async fn balance_checks(&self, user_id: u64) -> Web3ProxyResult<Arc<AtomicF64>> {
+        if user_id == 0 {
+            return Ok(Arc::new(AtomicF64::from(0.)));
+        }
+        match NonZeroU64::try_from(user_id) {
```

Comment:
this is redundant. the only time a NoneZeroU64 will error is if its 0. i'd delete 1106-08 and keep this match.

Reply:
done